### DISCUSSION TITLE
MicroPython snippets as part of Pymakr?

### DISCUSSION THREAD

**Discussion by Josverl** [pycom/pymakr-vsc#240] [Ideas]
It is quite easy to add useful micropython code snippets to Pymakr.
Add a reference in package.json and add another file with the snippets.

That would allow for a simple way to share reusable code fragments for both Pycom and other firmwares.
It would be useful to have an agreed way to identify specific firmware or hardware dependencies but I think that would be quite achievable.
Adding a code snippet requires just adding the snippet to the snippets.json file, and requires no code change. 

Alternatively other extensions can be used to separate this from pymakr,.

Do you think this is useful?
What types of code snippets or samples would you like to use?

**Comment by jakobrosenberg** [comment]
I'm very intrigued.

I see three options
A) Add the snippets directly to the package
B) Create the snippets in a separate package and make that package a dependency of pymakr-vsc
C) Create the snippets in a separate package and make an extension pack that includes pymakr-vsc and the new snippets package

A and B both suffer from tight coupling and we risk locking users to snippets which they might not want.
I was against C at first, but the more I think of it, the more I like the idea of a Pymakr extension pack as we could then provide a very agile setup for new users.

**Reply by Josverl** [reply]
Agree that **C** makes sense and allows things to be de-coupled.
Apart from an extension pack , I think/hope it should also be possible to add links in a walkthrough to a Snippet extension 

Using a pack would also allow users to install a pack with multiple sets of snippets, and then remove/disable the ones they do not want to use.
That is AFAIK a unique capability of the packs.

**Comment by jakobrosenberg** [comment]
> Apart from an extension pack , I think/hope it should also be possible to add links in a walkthrough to a Snippet extension

That shouldn't be a problem at all.
